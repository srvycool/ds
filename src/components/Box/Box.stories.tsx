import React from 'react';
import { Box } from './Box';
import { Typography } from '../Typography/Typography';
import { Button } from '../Button/Button';

export default {
  title: 'Box',
  component: Box,
};

export const All: React.FC = () => (
  <>
    <Box
      padding="xl"
      backgroundColor="primary100"
      borderRadius="primary"
      boxShadow="primary"
      marginBottom="m"
    >
      <Typography variant="heading1" color="grey900" marginBottom="s">
        Dolor irure anim qui in sit nulla
      </Typography>
      <Typography variant="body1" color="grey900">
        Quis sit pariatur culpa anim minim quis eiusmod pariatur veniam nostrud
        quis elit.
      </Typography>
    </Box>
    <Box
      display="flex"
      flexDirection={['column', 'row']}
      justifyContent="space-between"
      alignItems="center"
      padding="xl"
      backgroundColor="grey100"
      borderRadius="primary"
      boxShadow="primary"
      marginBottom="m"
    >
      <Typography
        variant="body1"
        color="grey900"
        marginBottom={['l', 'zero']}
        marginRight={['zero', 'l']}
        textAlign={['center', 'left']}
      >
        Quis sit pariatur culpa anim minim quis eiusmod pariatur veniam nostrud
        quis elit.
      </Typography>
      <Button>Button</Button>
    </Box>
    <Box
      display="flex"
      justifyContent="center"
      padding="xl"
      borderBottom="primary"
    >
      <Typography variant="heading1" color="grey900">
        Dolor irure anim qui in sit nulla
      </Typography>
    </Box>
  </>
);
