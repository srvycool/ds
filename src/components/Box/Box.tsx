import styled from 'styled-components';
import {
  compose,
  space,
  color,
  shadow,
  flexbox,
  layout,
  border,
  position,
  BackgroundColorProps,
  MarginBottomProps,
  ShadowProps,
  PaddingProps,
  BoxShadowProps,
  FlexboxProps,
  BorderProps,
  LayoutProps,
  PositionProps,
} from 'styled-system';
import { Theme } from '../../theme';

export type BoxProps = BackgroundColorProps<Theme> &
  MarginBottomProps<Theme> &
  ShadowProps<Theme> &
  PaddingProps<Theme> &
  BoxShadowProps<Theme> &
  FlexboxProps<Theme> &
  BorderProps<Theme> &
  LayoutProps<Theme> &
  PositionProps<Theme>;

export const Box = styled.div<BoxProps>`
  ${compose(color, space, shadow, flexbox, border, layout, position)}
  box-sizing: border-box;
`;
