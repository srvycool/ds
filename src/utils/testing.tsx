import React from 'react';
import {
  render as tlrender,
  RenderOptions,
  RenderResult,
} from '@testing-library/react';
import { ThemeProvider } from '../components/ThemeProvider/ThemeProvider';

export function render(
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'queries'>
): RenderResult {
  return tlrender(<ThemeProvider>{ui}</ThemeProvider>, options);
}
