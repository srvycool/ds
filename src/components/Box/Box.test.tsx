import React from 'react';
import { render } from '../../utils/testing';
import { Box } from './Box';

describe('Box', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <Box
        display="flex"
        flexDirection="row"
        justifyContent="space-between"
        alignItems="center"
        padding="xl"
        backgroundColor="grey100"
        borderRadius="primary"
        boxShadow="primary"
        marginBottom="m"
      >
        Sit eiusmod ullamco eiusmod non cillum proident.
      </Box>
    );

    expect(container).toMatchSnapshot();
  });
});
