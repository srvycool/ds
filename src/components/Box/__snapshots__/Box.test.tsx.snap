// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 40px;
  background-color: #F1ECED;
  border-radius: 8px;
  box-shadow: 0px 1px 2px rgba(0,0,0,0.06),0px 1px 3px rgba(0,0,0,0.1);
  margin-bottom: 16px;
  box-sizing: border-box;
}

<div>
  <div
    class="c0"
    display="flex"
  >
    Sit eiusmod ullamco eiusmod non cillum proident.
  </div>
</div>
`;
